poller/poller.go

```diff
@@ -173,44 +173,61 @@ func (poller *Poller) getSerializedDataSubset(splitNames []string) string {
 func (poller *Poller) getUpdatedSerializedDataSubsets(newSplitData SplitData) map[string]string {
 	updatedSubsets := poller.getCachedSerializedDataSubsets()
 	for key := range updatedSubsets {
-		updatedSubsets[key] = generateSerializedData(newSplitData, strings.Split(key, "."))
+		updatedSubsets[key] = poller.generateSerializedData(newSplitData, strings.Split(key, "."))
 	}
 	return updatedSubsets
 }
 
 // generateSerializedData takes SplitData and generates a script tag
 // that saves the SplitData info to the window object of the browser
-func generateSerializedData(splitData SplitData, splitNames []string) string {
+func (poller *Poller) generateSerializedData(splitData SplitData, splitNames []string) string {
 	if reflect.DeepEqual(splitData, SplitData{}) {
 		return emptyCacheLoggingScript
 	}
 	splitNamesToSerializedData := map[string]string{}
+	splitsSubset := map[string]dtos.SplitDTO{}
 
 	// Serialize values for splits
-	for _, split := range splitData.Splits {
-		index := sort.SearchStrings(splitNames, split.Name)
-		splitIsInSplitNames := index < len(splitNames) && splitNames[index] == split.Name
-		// if the split is not in the splitNames array, do not serialize the split
-		if len(splitNames) > 0 && !splitIsInSplitNames {
-			continue
+	for name, split := range splitData.Splits {
+		if len(splitNames) > 0 {
+			index := sort.SearchStrings(splitNames, split.Name)
+			splitIsInSplitNames := index < len(splitNames) && splitNames[index] == split.Name
```

Comment:
I wonder: will `splitNames[index] == split.Name` ever be false? when left part is true

Reply:
Yes. Based on [the documentation](https://golang.org/pkg/sort/#SearchStrings): "The return value is the index to insert x if x is not present". Thus, the index could be valid, but the string at that index could potentially not match if x is not in the array

Reply:
but wouldn't it be captured in the left part already? if x is not present, the index should = len(splitNames)

Reply:
Consider this case:
```go
index := sort.SearchStrings([]string{"a", "c", "d"}, "b")
fmt.Println("index: ", index)
```
Index is `1` in this case because `b` would be inserted at index 1, but the value at index 1 is `c`.

Reply:
oh okay, because it is sorted. thanks!